### DISCUSSION TITLE
Tiddlywiki as a Static Site Generator

### DISCUSSION THREAD

**Discussion by kookma** [TiddlyWiki/TiddlyWiki5#5586] [Ideas]
Hi Jeremy,

 I follow your work on [publishing framework branch](https://github.com/Jermolene/TiddlyWiki5/tree/publishing-framework)! Is there ay plan to have a release for testing purpose any time soon?
 Furthermore is there any room for comments, feature request?

**Comment by saqimtiaz** [comment]
@kookma I suspect it would be extremely helpful if we checked out that branch, tested it and provided feedback. I have been meaning to but have not found the time for it thus far.

**Comment by Jermolene** [comment]
> Hi Jeremy,
> 
> 
> 
>  I follow your work on [publishing framework branch](https://github.com/Jermolene/TiddlyWiki5/tree/publishing-framework)! Is there ay plan to have a release for testing purpose any time soon?
> 
>  Furthermore is there any room for comments, feature request?
> 
> 
> 
> 

Good point. We don't currently have continuous integration setup to publish branches, but I'll figure out a way to deploy a test build somewhere else, perhaps TiddlyHost.

**Comment by Jermolene** [comment]
I've uploaded a preview here:

https://tiddlywiki-pr-5559.tiddlyhost.com

Click the publish button in the sidebar to make an 18.5MB static site build and download it as a zip file.

<img width="34" alt="image" src="https://user-images.githubusercontent.com/174761/113596428-3401da80-9632-11eb-86f8-c98f8dbadf66.png">

**Reply by kookma** [reply]
Lovely! Many thanks Jeremy!
Should we send comments, feature request here in discussion forum?

**Reply by pmario** [reply]
The index.html in the ZIP file is a fully working TW. 

There is no static.html as a starting point in the "root" dir. .. So where is the "starting point"?

**Reply by Jermolene** [reply]
Hopefully you can see what's going on by looking at the sitemap. The one that's included just saves every non-system wikitext tiddler as a separate HTML file. At this point it's just reusing existing templates in the core. Extending it beyond that is hopefully just a matter of applying what you already know about static site generation on TiddlyWiki. This is a PRERELEASE, and is not complete. The things to focus on are the things that are new, which are hopefully all listed in the PR.

**Reply by Jermolene** [reply]
Best place for feedback is the pull request:

https://github.com/Jermolene/TiddlyWiki5/pull/5559
